tests/formats/format_helpers.test.ts

```diff
@@ -1022,3 +1094,131 @@ describe("create currency format", () => {
+    expect(formatValue(0, { locale, format })).toBe("0$");
+    expect(formatValue(-1, { locale, format })).toBe("1€");
+    expect(formatValue("text", { locale, format })).toBe("text");
+  });
+
+  test("Multi-part format with zero values format", () => {
+    const format = "0[$$];0[$€];0[$£]";
+
+    expect(formatValue(1, { locale, format })).toBe("1$");
+    expect(formatValue(0, { locale, format })).toBe("0£");
+    expect(formatValue(-1, { locale, format })).toBe("1€");
+    expect(formatValue("text", { locale, format })).toBe("text");
+  });
+
+  test("Multi-part format with text values format", () => {
+    const format = "0[$$];0[$€];0[$£];@[$¥]";
+
+    expect(formatValue(1, { locale, format })).toBe("1$");
+    expect(formatValue(0, { locale, format })).toBe("0£");
+    expect(formatValue(-1, { locale, format })).toBe("1€");
+    expect(formatValue("text", { locale, format })).toBe("text¥");
+  });
+
+  test("Multi part format with two parts", () => {
+    const format = "0[$$];0[$€]";
+
+    expect(formatValue(1, { locale, format })).toBe("1$");
+    expect(formatValue(0, { locale, format })).toBe("0$");
+    expect(formatValue(-1, { locale, format })).toBe("1€");
+    expect(formatValue("text", { locale, format })).toBe("text");
+  });
+
+  test("Multi part format with three parts", () => {
+    const format = "0[$$];0[$€];0[$£]";
+
+    expect(formatValue(1, { locale, format })).toBe("1$");
+    expect(formatValue(0, { locale, format })).toBe("0£");
+    expect(formatValue(-1, { locale, format })).toBe("1€");
+    expect(formatValue("text", { locale, format })).toBe("text");
+  });
+
+  test("Text-part of format cannot be a number or date format", () => {
+    expect(() => formatValue(1, { locale, format: ";;;0" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";;;dd" })).toThrow();
+  });
+
+  test("negative and null format cannot be text format", () => {
+    expect(() => formatValue(1, { locale, format: "@;;;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";@;;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";;@;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";;;@" })).not.toThrow();
+
+    expect(() => formatValue(1, { locale, format: "@;;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";@;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";;@" })).toThrow();
+
+    expect(() => formatValue(1, { locale, format: "@;" })).toThrow();
+    expect(() => formatValue(1, { locale, format: ";@" })).toThrow();
+
+    expect(() => formatValue(1, { locale, format: "@" })).not.toThrow();
+  });
+
+  test("Can make operations on multi part format", () => {
+    expect(roundFormat("0.00\\€;$0.#; 0.00 ;@")).toBe("0\\€;$0; 0 ;@");
+
+    expect(changeDecimalPlaces("0.0\\€;$0.#; 0 ;@", 1)).toBe("0.00\\€;$0.#0; 0.0 ;@");
+    expect(changeDecimalPlaces("0.0\\€;$0.#; 0 ;@", -1)).toBe("0\\€;$0; 0 ;@");
```

Comment:
Maybe add a test when remove decimal place with something between digits

ex `changeDecimalPlaces('0.0\\€0;$0.#"TEST"0; 0 ;@', -1))`

Reply:
and if we end up supporting this kind of writing on the text part `;;;@0` then testing that the remove of digit does not change anything ?